tests/SpecTests/ClientSideEncryption/Prose22_RangeExplicitEncryptionTest.php

```diff
@@ -0,0 +1,479 @@
+        self::insertKeyVaultData($client, [$key1Document]);
+
+        $this->clientEncryption = $client->createClientEncryption([
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+        ]);
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+            'bypassQueryAnalysis' => true,
+        ];
+
+        $this->encryptedClient = self::createTestClient(null, [], [
+            'autoEncryption' => $autoEncryptionOpts,
+            /* libmongocrypt caches results from listCollections. Use a new
+             * client in each test to ensure its encryptedFields is applied. */
+            'disableClientPersistence' => true,
+        ]);
+    }
+
+    public function setUpWithTypeAndRangeOpts(string $type, array $rangeOpts): void
+    {
+        if ($type === 'DecimalNoPrecision' || $type === 'DecimalPrecision') {
+            $this->markTestSkipped('Bundled libmongocrypt does not support Decimal128 (PHPC-2207)');
+        }
+
+        /* Read the encryptedFields file directly into BSON to preserve typing
+         * for 64-bit integers. This means that DropEncryptedCollection and
+         * CreateEncryptedCollection will be unable to inspect the option for
+         * metadata collection names, but that's not necessary for the test. */
+        $encryptedFields = Document::fromJSON(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/range-encryptedFields-' . $type . '.json'));
+
+        $database = $this->encryptedClient->selectDatabase($this->getDatabaseName());
+        $database->dropCollection('explicit_encryption', ['encryptedFields' => $encryptedFields]);
+        $database->createCollection('explicit_encryption', ['encryptedFields' => $encryptedFields]);
+        $this->collection = $database->selectCollection('explicit_encryption');
+
+        $encryptOpts = [
+            'keyId' => $this->key1Id,
+            'algorithm' => ClientEncryption::ALGORITHM_RANGE_PREVIEW,
+            'contentionFactor' => 0,
+            'rangeOpts' => $rangeOpts,
+        ];
+
+        $cast = self::getCastCallableForType($type);
+        $fieldName = 'encrypted' . $type;
+
+        $this->collection->insertMany([
+            ['_id' => 0, $fieldName => $this->clientEncryption->encrypt($cast(0), $encryptOpts)],
+            ['_id' => 1, $fieldName => $this->clientEncryption->encrypt($cast(6), $encryptOpts)],
+            ['_id' => 2, $fieldName => $this->clientEncryption->encrypt($cast(30), $encryptOpts)],
+            ['_id' => 3, $fieldName => $this->clientEncryption->encrypt($cast(200), $encryptOpts)],
+        ]);
+    }
+
+    public function tearDown(): void
+    {
+        $this->collection = null;
```

Comment:
We don't free the `Collection` instance in other test cases. Did you hit an issue?

Reply:
Since `encryptedClient` is created with `disableClientPersistence`, I originally did this to ensure that libmongoc's `mongoc_client_t` would be destroyed at the end of the test. This has been done before, but usually for session objects to ensure they are returned to the internal pool.

I just tested this locally and can confirm that without this, the Manager objects and `mongoc_client_t` instances accumulate and are not freed until the end of the test _suite_ (not just the test _case_). I opened [PHPLIB-1154](https://jira.mongodb.org/browse/PHPLIB-1154) to track this.

In the meantime, I've added a comment to explain this and removed the unsetting of `key1Id`, which wasn't necessary.